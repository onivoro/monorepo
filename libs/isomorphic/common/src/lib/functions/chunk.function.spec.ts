import { chunk } from './chunk.function';
import { arrangeActAssert } from '../functions/arrange-act-assert.function';

const array = ['one', 'two', 'three', 'four', 'five', 'six', 'seven', 'eight'];

describe('chunk', () => {
    it.each([
        [1],
        [2],
        [3],
        [4],
        [5],
    ])('given %j, returns an array of arrays', async (_) => await arrangeActAssert({
        arrange: () => ({ subject: chunk }),
        act: ({ subject }) => subject(array, _),
        assert: ({ result }) => {
            expect(result).toMatchSnapshot();
        }
    }));
});