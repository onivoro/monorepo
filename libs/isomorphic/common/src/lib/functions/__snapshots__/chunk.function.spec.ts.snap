// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`chunk given 1, returns an array of arrays 1`] = `
[
  [
    "one",
    "two",
    "three",
    "four",
    "five",
    "six",
    "seven",
    "eight",
  ],
]
`;

exports[`chunk given 2, returns an array of arrays 1`] = `
[
  [
    "one",
    "two",
    "three",
    "four",
  ],
  [
    "five",
    "six",
    "seven",
    "eight",
  ],
]
`;

exports[`chunk given 3, returns an array of arrays 1`] = `
[
  [
    "one",
    "two",
    "three",
  ],
  [
    "four",
    "five",
    "six",
  ],
  [
    "seven",
    "eight",
  ],
]
`;

exports[`chunk given 4, returns an array of arrays 1`] = `
[
  [
    "one",
    "two",
  ],
  [
    "three",
    "four",
  ],
  [
    "five",
    "six",
  ],
  [
    "seven",
    "eight",
  ],
]
`;

exports[`chunk given 5, returns an array of arrays 1`] = `
[
  [
    "one",
    "two",
  ],
  [
    "three",
    "four",
  ],
  [
    "five",
    "six",
  ],
  [
    "seven",
    "eight",
  ],
]
`;
